### ISSUE TITLE
Replace `git rev-parse HEAD` with `Utils.git_head`

### ISSUE THREAD

**Issue by SeekingMeaning** [Homebrew/homebrew-core#68805]
With the release of `brew` 2.7.3 and 2.7.4, formulae can now use the `Utils.git_head` helper function instead of `git rev-parse HEAD`

```rb
# Old:
Utils.safe_popen_read("git", "rev-parse", "HEAD").chomp
Utils.safe_popen_read("git", "rev-parse", "--short", "HEAD").chomp

# New:
Utils.git_head
Utils.git_short_head
```

Example PRs:
- https://github.com/Homebrew/homebrew-core/pull/69007/files
- https://github.com/Homebrew/homebrew-core/pull/69008/files
- https://github.com/Homebrew/homebrew-core/pull/69009/files

Formulae containing `rev-parse`:

- [x] `arangodb` https://github.com/Homebrew/homebrew-core/pull/69007
- [x] `arduino-cli` https://github.com/Homebrew/homebrew-core/pull/69008
- [x] `aws-iam-authenticator` - https://github.com/Homebrew/homebrew-core/pull/69009
- [x] `buildkit` #68808 + #68975
- [x] `cayley` https://github.com/Homebrew/homebrew-core/pull/69010
- [x] `chart-testing` https://github.com/Homebrew/homebrew-core/pull/69011
- [x] `chezmoi` https://github.com/Homebrew/homebrew-core/pull/69012
- [x] `circleci` https://github.com/Homebrew/homebrew-core/pull/69014
- [x] `consul-template` https://github.com/Homebrew/homebrew-core/pull/69016
- [x] `crystal` #68813
- [x] `docker-machine-driver-xhyve` https://github.com/Homebrew/homebrew-core/pull/69017
- [x] `docker` #68809
- [x] `faas-cli` https://github.com/Homebrew/homebrew-core/pull/69138
- [x] `fleet-cli` https://github.com/Homebrew/homebrew-core/pull/69139
- [x] `gcsfuse` https://github.com/Homebrew/homebrew-core/pull/69140
- [x] `git-flow` does not apply to this issue
- [x] `gitlab-runner` https://github.com/Homebrew/homebrew-core/pull/69141
- [x] `gitup` #69024
- [x] `golangci-lint` https://github.com/Homebrew/homebrew-core/pull/69142
- [x] `goofys` https://github.com/Homebrew/homebrew-core/pull/69143
- [x] `h2spec` #68804 + #68976
- [x] `inlets` https://github.com/Homebrew/homebrew-core/pull/69144
- [x] `juju` https://github.com/Homebrew/homebrew-core/pull/69145
- [x] `k3sup` https://github.com/Homebrew/homebrew-core/pull/69146
- [x] `kapacitor` https://github.com/Homebrew/homebrew-core/pull/69019
- [x] `krew` #69218
- [x] `ksync` #69219
- [x] `kustomize` #69220
- [x] `mage` #69221
- [x] `minio-mc` https://github.com/Homebrew/homebrew-core/pull/69332
- [x] `minio` https://github.com/Homebrew/homebrew-core/pull/69267
- [x] `mmctl` #69222
- [x] `ndenv` #69223
- [x] `nodenv` #69224
- [x] `octant` #69225
- [x] `rbenv` #68810 + #68977
- [x] `sync_gateway` #69226
- [x] `tile38` #69227

Additionally, the following formulae should have `stable.specs[:revision]` and `stable.specs[:tag]` replaced with `Utils.git_head` and `version` (so that they can be built from e.g. `master`/`main`):

- [x] `calicoctl` https://github.com/Homebrew/homebrew-core/pull/69018
- [x] `devspace` #69216
- [x] `gateway-go` #69217
- [x] `goreleaser` #69104
- [x] `k9s` #69103
- [x] `operator-sdk` #69180
- [x] `periscope` #69178
- [x] `server-go` #69181
- [x] `uptoc` #69182

**Comment by SeekingMeaning** [comment]
Closing this temporarily as `git_head` uses `popen_read` instead of `safe_popen_read`

cc @alebcay @carlocab

**Comment by alebcay** [comment]
> Closing this temporarily as git_head uses popen_read instead of safe_popen_read

@SeekingMeaning, is there a PR/commit in Homebrew/brew to track that fixes this?

**Comment by SeekingMeaning** [comment]
Also opened https://github.com/Homebrew/brew/pull/10306 so that we can just do `Utils.git_head` or `Utils.git_short_head`

**Comment by SeekingMeaning** [comment]
Both merged (and now available as of `brew` 2.7.4)

**Comment by SeekingMeaning** [comment]
Thanks for your help, everyone! 🚀
